pointgrey_camera_driver/src/nodelet.cpp

```diff
@@ -312,73 +273,188 @@ class PointGreyCameraNodelet: public nodelet::Nodelet
   */
   void devicePoll()
   {
+    enum State
+    {
+        NONE
+      , ERROR
+      , STOPPED
+      , DISCONNECTED
+      , CONNECTED
+      , STARTED
+    };
+
+    State state = DISCONNECTED;
```

Comment:
Only `connectCb` starts/call `devicePoll`, so I can assume the initial state is always `DISCONNECTED`. For that reason I have the `enum` and `state` variable local.